src/main.cpp

```diff
@@ -1007,12 +1037,19 @@ bool ContextualCheckTransaction(
                         REJECT_INVALID, "bad-cb-output-desc-invalid-encct");
                 }
 
+                auto leadByte = encPlaintext->get_leadbyte();
+                assert(leadByte == 0x01 || leadByte == 0x02);
+                zip_212_enabled = (leadByte == 0x02);
+
+                zaddr = libzcash::SaplingPaymentAddress(encPlaintext->d, outPlaintext->pk_d);
```

Comment:
I don't see an equivalent line setting `zaddr` in the other branch of the `if`; if this is intended please leave a comment reflecting why it's okay to check against the value provided by the default constructor.

Reply:
Oh, never mind, `std::tie`